presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/StaticHiveCluster.java

```diff
@@ -67,16 +71,25 @@ public HiveMetastoreClient createMetastoreClient()
         List<HostAndPort> metastores = new ArrayList<>(addresses);
         Collections.shuffle(metastores.subList(1, metastores.size()));
 
-        TTransportException lastException = null;
+        TException lastException = null;
         for (HostAndPort metastore : metastores) {
             try {
-                return clientFactory.create(metastore);
+                HiveMetastoreClient hiveMetastoreClient = clientFactory.create(metastore);
+                if (!isNullOrEmpty(hiveUserName)) {
+                    hiveMetastoreClient.setUGI(hiveUserName, new ArrayList<String>());
+                }
+                return hiveMetastoreClient;
             }
-            catch (TTransportException e) {
+            catch (TException e) {
                 lastException = e;
             }
         }
 
+        if (!isNullOrEmpty(hiveUserName)) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, format("Failed connecting to Hive metastore: %s , HiveUserName is %s ",
```

Comment:
I don't think we should expose the username here. This is a security configuration detail, and the value is in the configuration file (it's always the same), so the administrator won't have a problem looking it up.

Reply:
ok,I will fix it